File: Languages/Ruby/SelectionSort.rb
#!/usr/bin/env ruby

# Source: ChatGPT

$LOGGING = false

def selectionsort(arr)
  n = arr.length

  # Iterate through each element in the array
  for i in 0...n
    # Find the minimum element in the unsorted part of the array
    min_idx = i
    for j in i+1...n
      if arr[j] < arr[min_idx]
        min_idx = j
      end
    end

    # Swap the minimum element with the first element in the unsorted part of the array
    arr[i], arr[min_idx] = arr[min_idx], arr[i]
  end

  # Return the sorted array
  return arr
end


def main()
  arr = [10650,56341,77676,47158,34999,22009,19680,41870,31090,30668]
  ans = selectionsort(arr)
  if $LOGGING
    puts "Sorted array: #{ans}"
  else
    puts ""
  end
end

main()

File: Languages/JavaScript/SelectionSort.js
#!/usr/bin/env node

var LOGGING = false;

function selectionSort(arr) {
    let len = arr.length;
    for (let i = 0; i < len; i++) {
        let min = i;
        for (let j = i + 1; j < len; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        if (i !== min) {
            swap(arr, i, min);
        }
    }
    return arr;
}

function swap(arr, i, j) {
    let temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

function main() {
    let arr = [10650,56341,77676,47158,34999,22009,19680,41870,31090,30668]
    selectionSort(arr);
    if (LOGGING) {
        console.log("Sorted Array: " + arr);
    } else {
        console.log();
    }
}

main();
